src/emqx_shared_sub.erl

```diff
@@ -201,5 +260,7 @@ update_stats(State) ->
 
 %% erlang:is_process_alive/1 is expensive
 %% and does not work with remote pids
```

Comment:
`is_process_alive` has a receive clause to receive result as a message delivery, so I thought it would be slow.
just did a simple test from an Erlang shell, `is_process_alive` is actually slightly faster than ets lookup.
will change the implementation to: do not insert local pids, and call `is_process_alive` for local pids.